lib/timeline/component/item/BackgroundItem.js

```diff
@@ -105,16 +99,73 @@ BackgroundItem.prototype.redraw = function() {
     // update class
     var className = (this.data.className ? (' ' + this.data.className) : '') +
         (this.selected ? ' vis-selected' : '');
-    dom.box.className = this.baseClassName + className;
+    this.dom.box.className = this.baseClassName + className;
+  }
+}
+BackgroundItem.prototype._getDomComponentsSizes = function() {
+  // determine from css whether this box has overflow
+  this.overflow = window.getComputedStyle(this.dom.content).overflow !== 'hidden';
+  return {
+    content: {
+      width: this.dom.content.offsetWidth
+    }
+  }
+}
+BackgroundItem.prototype._updateDomComponentsSizes = function(sizes) {
+  // recalculate size
+  this.props.content.width = sizes.content.width;
+  this.height = 0; // set height zero, so this item will be ignored when stacking items
 
-    // determine from css whether this box has overflow
-    this.overflow = window.getComputedStyle(dom.content).overflow !== 'hidden';
+  this.dirty = false;
+}
 
-    // recalculate size
-    this.props.content.width = this.dom.content.offsetWidth;
-    this.height = 0; // set height zero, so this item will be ignored when stacking items
+BackgroundItem.prototype._repaintDomAdditionals = function() {
+}
 
-    this.dirty = false;
+/**
+ * Repaint the item
+ * @param {boolean} [returnQueue=false]  return the queue
+ * @return {boolean} the redraw queue if returnQueue=true
+ */
+BackgroundItem.prototype.redraw = function(returnQueue) {
+  var sizes
+  var queue = [
+    // create item DOM
+    this._createDomElement.bind(this),
```

Comment:
Ah, right. Now the new methods above come together. This is a definitive improvement (wrt readability).